src/parser/mod.rs

```diff
@@ -11516,7 +11524,10 @@ impl<'a> Parser<'a> {
                 }
             } else {
                 None
-            };
+            }
+        } else {
+            None
```

Comment:
would it make sense to unnest the logic with something like?
```rust
if self.parse_keywords([Keyword::DISTINCT, Keyword::AS, Keyword::VALUE]) {
    Some()
} else if self.parse_keywords([Keyword::DISTINCT, Keyword::AS, Keyword::STRUCT]) {
    Some()
} else if self.parse_keywords([Keyword::AS, Keyword::VALUE]) {
    Some()
} else if self.parse_keywords([Keyword::AS, Keyword::STRUCT]) {
    Some()
} else {
    None
};
```
Also maybe we can move the logic to its own method now like `let value_table_mode = self.parse_value_table_mode()`, since its no longer trivial